packages/core/epoch/types.go

```diff
@@ -246,38 +249,42 @@ func (al *NodesActivityLog) Bytes() []byte {
 	return objBytes
 }
 
-// endregion ///////////////////////////////////////////////////////////////////////////////////////////////////////////
+func NewNodesActivityLog() *NodesActivityLog {
+	return &NodesActivityLog{*shrinkingmap.New[Index, *ActivityLog]()}
+}
+
+// endregi`azon ///////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 // region ActivityLog //////////////////////////////////////////////////////////////////////////////////////////////////
 
 // ActivityLog is a time-based log of node activity. It stores information when a node is active and provides
 // functionality to query for certain timeframes.
 type ActivityLog struct {
-	SetEpochs *set.AdvancedSet[identity.ID] `serix:"0,lengthPrefixType=uint32"`
+	setEpochs *set.AdvancedSet[identity.ID] `serix:"0,lengthPrefixType=uint32"`
```

Comment:
Can we embed this set directly by removing its name as it is the only field? So functions such as ForEach, Add and so on will be exposed on ActivityLog itself, without the need to rewrite them.